File: python3/Two Pointers/triplet_sum_brute_force.py
from typing import List


def triplet_sum_brute_force(nums: List[int]) -> List[List[int]]:
    n = len(nums)
    # Use a hash set to ensure we don't add duplicate triplets.
    triplets = set()
    # Iterate through the indexes of all triplets.
    for i in range(n):
        for j in range(i + 1, n):          
            for k in range(j + 1, n):
                if nums[i] + nums[j] + nums[k] == 0:
                    # Sort the triplet before including it in the 
                    # hash set.
                    triplet = tuple(sorted([nums[i], nums[j], nums[k]]))
                    triplets.add(triplet)
    return [list(triplet) for triplet in triplets]

File: swift/Two Pointers/TripletSumBruteForce.swift
func tripletSumBruteForce(_ nums: [Int]) -> [[Int]] {
    let n = nums.count
    // Use a hash set to ensure we don't add duplicate triplets.
    var triplets = Set<[Int]>()
    // Iterate through the indexes of all triplets.
    for i in 0..<n {
        for j in i + 1..<n {
            for k in j + 1..<n {
                if nums[i] + nums[j] + nums[k] == 0 {
                    // Sort the triplet before including it in the 
                    // hash set.
                    let triplet = [nums[i], nums[j], nums[k]].sorted()
                    triplets.insert(triplet)
                }
            }
        }
    }
    return Array(triplets)
}